sa/sa.go

```diff
@@ -574,8 +579,16 @@ func (ssa *SQLStorageAuthority) MarkCertificateRevoked(ctx context.Context, seri
 func (ssa *SQLStorageAuthority) UpdateRegistration(ctx context.Context, reg core.Registration) error {
 	const query = "WHERE id = ?"
 	model, err := selectRegistration(ssa.dbMap.WithContext(ctx), query, reg.ID)
-	if err == sql.ErrNoRows {
-		return berrors.NotFoundError("registration with ID '%d' not found", reg.ID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return berrors.NotFoundError("registration with ID '%d' not found", reg.ID)
+		}
+		if strings.HasPrefix(err.Error(), "Error 1062: Duplicate entry") {
```

Comment:
This error comes from a SELECT. When would it return `Duplicate entry`?

Reply:
Whoops, it wouldn't. This is intended for the `Update` below.